03_Modules/SmartCharging/src/module/module.ts

```diff
@@ -228,12 +233,24 @@ export class SmartChargingModule extends AbstractModule {
   }
 
   @AsHandler(CallAction.ReportChargingProfiles)
-  protected _handleReportChargingProfiles(
+  protected async _handleReportChargingProfiles(
     message: IMessage<ReportChargingProfilesRequest>,
     props?: HandlerProperties,
-  ): void {
+  ): Promise<void> {
     this._logger.debug('ReportChargingProfiles received:', message, props);
 
+    const chargingProfiles = message.payload
+      .chargingProfile as ChargingProfileType[];
+    for (const chargingProfile of chargingProfiles) {
+      await this._chargingProfileRepository.createOrUpdateChargingProfile(
```

Comment:
Take a look at K09 the `Remark`:

> The charging profiles report can be split over multiple ReportChargingProfilesRequest messages, this can be because charging profiles for different charging sources need to be reported, or because there is just to much data for one message. To indicate that more reports will follow the flag tbc can be used.

 I think we might run into a problem here if the charging profile is split over multiple messages. (`tbc` is set `true`) in the message. It will overwrite the existing db entry. 
A solution direction split on if `tbc` in the db persisted object  is true and then merge/extend/stack incoming with saved in DB and if the income one has `tbc` false, be sure to persist this.

Reply:
> Take a look at K09 the `Remark`:
> 
> > The charging profiles report can be split over multiple ReportChargingProfilesRequest messages, this can be because charging profiles for different charging sources need to be reported, or because there is just to much data for one message. To indicate that more reports will follow the flag tbc can be used.
> 
> I think we might run into a problem here if the charging profile is split over multiple messages. (`tbc` is set `true`) in the message. It will overwrite the existing db entry. A solution direction split on if `tbc` in the db persisted object is true and then merge/extend/stack incoming with saved in DB and if the income one has `tbc` false, be sure to persist this.

I might have a different understanding on the remark. The charger stores a list of charging profiles. A single charging profile object will not be split.  But the size of charging profile list can be too large to send them all in a single ReportChargingProfilesRequest or charging profiles are from different sources. In these 2 cases, charger can split and send part of charging profiles in one ReportChargingProfilesRequest and left ones in other ReportChargingProfilesRequests.

Reply:
Ok. I dove in again and I think I get it now. So because of this constraint: `unique: 'stationId_evseId_id',` on the database object and that we save each profile with context of its `chargingLimitSource`, we know that it won't be overridden.